### PR TITLE
25 correct clippy

### PR DESCRIPTION
**Pull request by AugustoFKL**
# Description

- Updating the code to follow the Clippy good practices. Used the command `cargo clippy --all-targets --all-features`

Fixes part of #25

## Type of change
- [X] Refactor

### CODE REVIEW

**Comment by AugustoFKL** [comment]
@Samyak2 @psiayn sorry, can't ask review directly. If/when possible, please review!

File: src/ast.rs [hunk 1]
```diff
@@ -93,16 +93,6 @@ impl PartialEq for ExprKind {
...
-
-    fn ne(&self, other: &Self) -> bool {
-        match (self, other) {
-            (Self::Atom(l0), Self::Atom(r0)) => l0 != r0,
-            (Self::List(l0), Self::List(r0)) => l0 != r0,
-            (Self::Func(l0), Self::Func(r0)) => *l0 as usize != *r0 as usize,
-            (Self::Lambda(l0), Self::Lambda(r0)) => l0 != r0,
-            _ => false,
-        }
-    }
```
Comment (Samyak2) [inline src/ast.rs:105]: Why was this removed?

Comment (Samyak2) [reply src/ast.rs:105]: nvm, found it - https://rust-lang.github.io/rust-clippy/master/index.html#partialeq_ne_impl

File: src/eval/functions.rs [hunk 2]
```diff
@@ -95,11 +95,12 @@ pub fn execute_lambda(
             }
 
             // return the last Ok() result (will be last expr if everything succeeded), or
+
```
Comment (Samyak2) [inline src/eval/functions.rs]: What's this extra new line for?

File: benches/spresso_benchmark.rs [hunk 3]
```diff
@@ -1,42 +1,42 @@
-use criterion::{criterion_group, criterion_main, Criterion, BenchmarkId};
```
Comment (Samyak2) [inline benches/spresso_benchmark.rs:1]: Would be better to comment out the whole file for now, instead of just the function bodies

**Review by Samyak2 — CHANGES_REQUESTED** [review]
Looks great, thanks!

A few comments though.

File: src/ast.rs (hunk 1 above)
Comment (AugustoFKL) [reply src/ast.rs:105]: Will mark as resolved. It is exactly the Partial Eq thing

File: src/eval/functions.rs (hunk 2 above)
Comment (AugustoFKL) [reply src/eval/functions.rs]: Unnecessary, will remove it!

**Comment by AugustoFKL** [comment]
@Samyak2 done!

**Review by Samyak2 — APPROVED** [review]
LGTM. Thank you for contributing! :heart:

**Comment by Samyak2** [comment]
Looks like 6a75349ff403537ce122b27d88230d314144450b also fixed formatting (a part of #23), but that's ok

**Pull request merged**